src/Chapter3.hs

```diff
@@ -971,6 +1088,33 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek 
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday 
+  deriving(Show, Eq, Ord, Bounded, Enum)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend day = 
+  if day == Saturday || day == Sunday
+    then True
+    else False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay day = 
+  if day == Sunday
+    then Monday 
+    else succ day 
+
+daysToParty :: DayOfWeek -> Int
+daysToParty day
+  | day == Friday = 0
+  | day < Friday = length (enumFromTo day Friday) - 1
+  | day > Friday = (2 - length (enumFromTo Friday day)) + 6
```

Comment:
This could be done in a single line without all these cases with the use of `mod` function 🙂